MomonGA: added support for momonga source

peakedshout opened the pull request:
Added support for momonga source

closes #8988

Checklist:

- [x] Updated `extVersionCode` value in `build.gradle` for individual extensions
- [ ] Updated `overrideVersionCode` or `baseVersionCode` as needed for all multisrc extensions
- [x] Referenced all related issues in the PR body (e.g. "Closes #xyz")
- [x] Added the `isNsfw = true` flag in `build.gradle` when appropriate
- [ ] Have not changed source names
- [ ] Have explicitly kept the `id` if a source's name or language were changed
- [x] Have tested the modifications by compiling and running the extension through Android Studio
- [x] Have removed `web_hi_res_512.png` when adding a new extension

peakedshout committed:
New:
Added support for momonga source

peakedshout committed:
Merge branch 'main' into f_momonga_v1.4.1

peakedshout committed:
New: Added support for momonga source

peakedshout committed:
New: Added support for momonga source

peakedshout committed:
New: Added support for momonga source

peakedshout committed:
New: Added support for momonga source

vetleledaal commented on src/ja/momonga/res/mipmap-xxxhdpi/ic_launcher.png:
Use the icon generator mentioned in the contributing guidelines. You can use:
[ic_launcher.zip](https://github.com/user-attachments/files/20637782/ic_launcher.zip)

Preview:
![ic_launcher](https://github.com/user-attachments/assets/de9d3023-fb4b-4e70-a6df-32fd0a455cf2)

vetleledaal commented on src/ja/momonga/src/eu/kanade/tachiyomi/extension/ja/momonga/MomonGA.kt:
```diff
@@ -0,0 +1,169 @@
+    private val dateFormat = SimpleDateFormat("yyyy年M月d日H時", Locale.ENGLISH)
+
+    override fun chapterListParse(response: Response): List<SChapter> {
+        val doc = response.asJsoup()
```
By convention:
```suggestion
        val document = response.asJsoup()
```

vetleledaal commented on src/ja/momonga/src/eu/kanade/tachiyomi/extension/ja/momonga/MomonGA.kt:
```diff
@@ -0,0 +1,169 @@
+
+    override fun latestUpdatesParse(response: Response): MangasPage {
+        return MangasPage(emptyList(), false)
+    }
```
When a function is not used you should use:
```kt
throw UnsupportedOperationException()
```

vetleledaal commented on src/ja/momonga/src/eu/kanade/tachiyomi/extension/ja/momonga/MomonGA.kt:
```diff
@@ -0,0 +1,169 @@
+            }
+        }
+
+        return SManga.create().apply {
```
Can you move this line to before `doc.select("#post-tag...`? Then you should be able to avoid using `var` variables by setting the values directly.

vetleledaal commented on src/ja/momonga/src/eu/kanade/tachiyomi/extension/ja/momonga/MomonGA.kt:
```diff
@@ -0,0 +1,169 @@
+            author = dauthor
+            artist = dartist
+            genre = dgenre
+            thumbnail_url = doc.selectFirst("#post-hentai > img")!!.attr("src")
```
```suggestion
            thumbnail_url = doc.selectFirst("#post-hentai > img")?.absUrl("src")
```

vetleledaal commented on src/ja/momonga/src/eu/kanade/tachiyomi/extension/ja/momonga/MomonGA.kt:
```diff
@@ -0,0 +1,169 @@
+
+            when (div.selectFirst("div.post-tag-title")!!.text()) {
+                "サークル" -> {
+                    dartist = div.select("div.post-tags > a").joinToString(", ") { it.text() }
```
`, ` is the default
```suggestion
                    dartist = div.select("div.post-tags > a").joinToString { it.text() }
```

vetleledaal commented on src/ja/momonga/src/eu/kanade/tachiyomi/extension/ja/momonga/MomonGA.kt:
```diff
@@ -0,0 +1,169 @@
+        val doc = response.asJsoup()
+
+        doc.select("#post-hentai > img").forEachIndexed { index, element ->
+            add(Page(index, "", element.attr("src")))
```
```suggestion
            add(Page(index, imageUrl = element.absUrl("src")))
```

vetleledaal commented on src/ja/momonga/src/eu/kanade/tachiyomi/extension/ja/momonga/MomonGA.kt:
```diff
@@ -0,0 +1,169 @@
+        val doc = response.asJsoup()
+
+        val lis = mutableListOf<SManga>()
+        doc.select("div.post-list > a").forEach { element ->
```
Consider using a `ParsedHttpSource` instead. You get some logic, like this, for free.

peakedshout replied:
No, this ParsedHttpSource has some override methods that confuse me. And it is more intuitive to write code using HttpSource.
If it is not mandatory, please allow me to continue using HttpSource.

vetleledaal commented on src/ja/momonga/src/eu/kanade/tachiyomi/extension/ja/momonga/MomonGA.kt:
```diff
@@ -0,0 +1,169 @@
+                SManga.create().apply {
+                    title = element.select("span").text()
+                    url = element.attr("href").toHttpUrl().encodedPath
+                    thumbnail_url = element.selectFirst("div.post-list-image > img")?.attr("src")
```
```suggestion
                    thumbnail_url = element.selectFirst("div.post-list-image > img")?.absUrl("src")
```

vetleledaal commented on src/ja/momonga/src/eu/kanade/tachiyomi/extension/ja/momonga/MomonGA.kt:
```diff
@@ -0,0 +1,169 @@
+        doc.select("div.post-list > a").forEach { element ->
+            lis.add(
+                SManga.create().apply {
+                    title = element.select("span").text()
```
You only expect one element here, probably.
```suggestion
                    title = element.selectFirst("span")!!.text()
```

vetleledaal commented on src/ja/momonga/src/eu/kanade/tachiyomi/extension/ja/momonga/MomonGA.kt:
```diff
@@ -0,0 +1,169 @@
+            lis.add(
+                SManga.create().apply {
+                    title = element.select("span").text()
+                    url = element.attr("href").toHttpUrl().encodedPath
```
Usually you would use `setUrlWithoutDomain(element.absUrl("href"))`. Both `setUrlWithoutDomain` and `absUrl` to give Jsoup the responsibility of resolving relative URLs.

vetleledaal commented on src/ja/momonga/src/eu/kanade/tachiyomi/extension/ja/momonga/MomonGA.kt:
```diff
@@ -0,0 +1,169 @@
+        return MangasPage(lis, doc.selectFirst("div.wp-pagenavi > a.nextpostslink") != null)
+    }
+
+    override fun popularMangaRequest(page: Int): Request = GET("$baseUrl/popularity/", headers)
```
How do you handle pagination?

peakedshout replied:
What mean?

peakedshout replied:
``doc.selectFirst("div.wp-pagenavi > a.nextpostslink") != null``

This is a component that detects whether there is a next page and is used to determine whether there are subsequent pages.
After testing, this is feasible.

vetleledaal replied:
I mean the URL does not contain the page number, so you will be requesting the same page (first page) when the app requests page 2, 3, 4, etc.

vetleledaal commented on src/ja/momonga/src/eu/kanade/tachiyomi/extension/ja/momonga/MomonGA.kt:
```diff
@@ -0,0 +1,169 @@
+        }
+
+        val url: String = if (query != "" && !query.contains("-")) {
+            urlBuilder.encodedQuery("s=$query").build().toString()
```
Use `.addQueryParameter()`

peakedshout committed:
Merge branch 'main' into f_momonga_v1.4.1

peakedshout committed:
Merge branch 'main' into f_momonga_v1.4.1

peakedshout committed:
Manwa: Optimize implementation logic (#9103)

* Optimization:
1. Dynamically obtain the list of optional image sources;
2. Dynamically obtain the redirected access URL;
3. Reconstruct some access logic;
4. Support filtering and next page functions;

* Optimization:
1. Dynamically obtain the list of optional image sources;
2. Dynamically obtain the redirected access URL;
3. Reconstruct some access logic;
4. Support filtering and next page functions;

* Optimization:
1. Dynamically obtain the list of optional image sources;
2. Dynamically obtain the redirected access URL;
3. Reconstruct some access logic;
4. Support filtering and next page functions;

* Optimization:
1. Dynamically obtain the list of optional image sources;
2. Dynamically obtain the redirected access URL;
3. Reconstruct some access logic;
4. Support filtering and next page functions;

kyrie25 committed:
Add MayoTune (#9106)

* Add MayoTune

* Use fallback for text contents

* Use DTO and refactor

* Use `selectFirst` for status

* Update fallback thumbnail URL

* Lint code

* Implement as single source

* Encapsulate `ChapterDto`

* Use relative URL

* Correctly handle request endpoints

peakedshout committed:
New: Added support for rumanhua source (#9128)

* New: Added support for rumanhua source

* New: Added support for rumanhua source

* New: Added support for rumanhua source

* New: Added support for rumanhua source

choppeh committed:
WeebCentral: add support for deeplink (#9129)

* Add support for deeplink

* Bump version

* Refactoring

* Add suggested changes

* Remove empty search

choppeh committed:
ArgosScan: Fixes (#9135)

* Fix

* Use relative manga url

Nyantad committed:
fix (Poseidonscans/PhenixScans) : Update Poseidon Scans and Phenix Scans (#9143)

* fix (PhenixScans): correct API base URL and improve chapter naming format in PoseidonScans

* update version codes

Nyantad committed:
[multisrc/pizzareader] Fix: Handle null status from API (#9144)

update PizzaReader to handle comic status correctly

AwkwardPeak7 commented on src/ja/momonga/src/eu/kanade/tachiyomi/extension/ja/momonga/MomonGA.kt:
```diff
@@ -0,0 +1,158 @@
+            urlBuilder.addQueryParameter("s", query).build().toString()
+        } else {
+            val path = filters.filterIsInstance<UriPartFilter>().joinToString("") { it.toUriPart() }
+            urlBuilder.encodedPath(path).build().toString()
```
won't this replace the `/page/$page` path?

AwkwardPeak7 commented on src/ja/momonga/src/eu/kanade/tachiyomi/extension/ja/momonga/MomonGA.kt:
```diff
@@ -0,0 +1,158 @@
+        }
+
+        val url: String = if (query != "" && !query.contains("-")) {
+            urlBuilder.addQueryParameter("s", query).build().toString()
```
use `.build()` instead of `.toString()`, as GET can take `HttpUrl`

JakeeLuwi committed:
Readcomiconline - Update config.json (#9150)

Update config.json

JakeeLuwi committed:
Readcomiconline - Update config.json (#9169)

Update config.json

peakedshout committed:
Merge branch 'main' into f_momonga_v1.4.1
